request.go

```diff
@@ -191,8 +191,18 @@ func (s *Server) handleConnect(ctx context.Context, conn conn, req *Request) err
 	defer target.Close()
 
 	// Send success
-	local := target.LocalAddr().(*net.TCPAddr)
-	bind := AddrSpec{IP: local.IP, Port: local.Port}
+	laddr := target.LocalAddr()
+	switch local := laddr.(type) {
+	case *net.TCPAddr, *net.UDPAddr:
+		bind.IP = local.IP
+		bind.Port = local.Port
+		break
+	case *net.IPAddr:
+		bind.IP = local.IP
```

Comment:
@myleshorton We aren't using that. We use a custom dial function in flashlight when we're configuring the socks server: https://github.com/getlantern/flashlight/blob/9655679e564389d0fdcca0d43a713b0754d6e4ee/client/client.go#L168